File: midlab3/MP03.js
const fs = require('fs');
const readline = require('readline');

// CSV file path
const filePath = "Sample_Data-Prog-2-csv.csv";

// Create readline interface for user input
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// Function to search CSV
function searchCSV(keyword) {
    return new Promise((resolve, reject) => {
        if (!fs.existsSync(filePath)) {
            console.log("CSV file not found.");
            resolve();
            return;
        }

        let foundCount = 0;
        const fileStream = fs.createReadStream(filePath);
        const reader = readline.createInterface({ input: fileStream });

        reader.on('line', (line) => {
            line = line.trim();
            if (line === "" || line.startsWith("Candidate")) return;

            if (line.toLowerCase().includes(keyword.toLowerCase())) {
                console.log(line);
                foundCount++;
            }
        });

        reader.on('close', () => {
            if (foundCount === 0) {
                console.log(`No results found for "${keyword}".`);
            }
            resolve();
        });
    });
}

// Loop function for repeated searches
async function loopSearch() {
    while (true) {
        const keyword = await new Promise((res) => {
            rl.question("\nEnter keyword to search (or type 'exit' to quit): ", res);
        });

        if (keyword.trim().toLowerCase() === "exit") {
            console.log("Exiting program.");
            rl.close();
            break;
        }

        await searchCSV(keyword);
    }
}

// Start the loop
loopSearch();

File: midlab3/MP03.java
package midlab3;

import java.io.*;
import java.util.Scanner;

public class MP03 {
    public static void main(String[] args) {
        String filePath = "Sample_Data-Prog-2-csv.csv";
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter keyword to search: ");
        String keyword = scanner.nextLine().toLowerCase();

        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
            String line;
            int foundCount = 0;

            while ((line = br.readLine()) != null) {
                line = line.trim();
                if (line.isEmpty() || line.startsWith("Candidate")) continue;

                if (line.toLowerCase().contains(keyword)) {
                    System.out.println(line);
                    foundCount++;
                }
            }

            if (foundCount == 0) {
                System.out.println("No results found for \"" + keyword + "\".");
            }

        } catch (IOException e) {
            System.out.println("Error reading file: " + e.getMessage());
        }
    }
}